pages/example-4-missing-authorization-gssp/common.tsx

```diff
@@ -47,19 +47,22 @@ export default function Common(props: Props) {
       <hr />
 
       <div>
-        The following blog posts should only be visible for logged-in users.
+        Below, a list of unpublished blog posts will appear for logged-in users
+        - similar to a "Drafts" list in a CMS.
         <br />
         <br />
-        If a user is not logged in, an error message should appear.
+        Each unpublished blog post should only be visible for the owner of the
+        post.
       </div>
 
-      <h2>Blog Posts</h2>
+      <h2>Unpublished Blog Posts</h2>
 
       {'error' in props && <div style={{ color: 'red' }}>{props.error}</div>}
 
       {'blogPosts' in props &&
         props.blogPosts.map((blogPost) => {
           return (
+            ( typeof props.userObject != 'undefined' && props.userObject.id == blogPost.userId) &&
```

Comment:
Yeah let's still split this into the vulnerable-1.tsx and vulnerable-2.tsx files, so that we don't have the complexity here

Reply:
ok.. this is a little trickier than I thought if we want to have the same commons page for vuln-1 and vuln-2 and the client-side check in vuln-2.
Because the getServerSideProps function of vuln2 runs on the server-side, so the only place to put the client-side control is in the "MissingAuthentication" function. But its not possible to just modify the content of props.blogPosts since props need to be derived from parent components as far as I understood...
Or is thera any trick to make that work?
I mean, we can also just skip the client-side control here - we already have it at example 3
what do you think?

Reply:
My original idea was what I wrote you offline:

> You could combine `vulnerable.tsx` and `common.tsx` to create this new file

But no worries, I can take over from here and finish this PR